CMakeLists.txt

```diff
@@ -875,9 +875,16 @@ else ()
 endif ()
 
 if (Seastar_DPDK)
+  # Although Seastar uses DPDK internally, DPDK symbols should not be exposed to
+  # Seastar users. However, DPDK's dependencies (like numa, pthread) must still
+  # be included in Seastar's INTERFACE_LINK_LIBRARIES to ensure proper linking
+  # of applications using Seastar.
   target_link_libraries (seastar
     PRIVATE
-      DPDK::dpdk)
+      "$<BUILD_INTERFACE:DPDK::dpdk>")
+  get_target_property(DPDK_INTERFACE_LINK_LIBRARIES DPDK::dpdk INTERFACE_LINK_LIBRARIES)
+  target_link_libraries (seastar
+    PRIVATE "${DPDK_INTERFACE_LINK_LIBRARIES}")
```

Comment:
could you use something like https://github.com/scylladb/seastar/blob/665fed057d6ce08de4a78659403203bcce79b6cb/CMakeLists.txt#L867-L875 ?

so we don't forget dropping the workaround for older versions of CMake when we bump up the minimal required version of CMake ?